devicetypes/qubino/qubino-flush-shutter.src/qubino-flush-shutter.groovy

```diff
@@ -0,0 +1,501 @@
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "unable to extract secure command from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, ep = null) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd, ep = null) {
+	log.debug "SwitchMultilevelReport ${cmd} from endpoint: ${ep}"
+	if (cmd.value != 0xFE) {
+		if (ep != 2) {
+			shadeEvent(cmd.value)
+		} else {
+			def event = [name: "level", value: cmd.value != 0x63 ? cmd.value : 100]
+			sendEventsToVenetianBlind([event])
+		}
+	} else {
+		log.warn "Something went wrong with calibration, position of blind is unknown"
+		if (ep == 2) {
+			sendEventsToVenetianBlind([[name: "level", value: 0]])
+		} else {
+			[
+					createEvent([name: "windowShade", value: "unknown"]),
+					createEvent([name: "level", value: 0]),
+					createEvent([name: "shadeLevel", value: 0])
+			]
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelSet cmd, ep = null) {
+	sendHubCommand(encap(zwave.meterV3.meterGet(scale: 0x02)))
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, ep = null) {
+	log.debug "BasicReport ${cmd}"
+	if (cmd.value != 0xFE && ep != 2) {
+		shadeEvent(cmd.value)
+	} else {
+		log.warn "Something went wrong with calibration, position of blind is unknown"
+	}
+}
+
+private shadeEvent(value) {
+	def shadeValue
+	def events = []
+	if (!value) {
+		shadeValue = "open"
+	} else if (value == 0x63) {
+		shadeValue = "closed"
+	} else {
+		shadeValue = "partially open"
+	}
+	events += createEvent([name: "windowShade", value: shadeValue])
+	events += createEvent([name: "level", value: value != 0x63 ? value : 100])
+	events += createEvent([name: "shadeLevel", value: value != 0x63 ? value : 100])
+
+	events
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, ep = null) {
+	def events = []
+	if (cmd.meterType == 0x01) {
+		def eventMap = [:]
+		if (cmd.scale == 0x00) {
+			eventMap.name = "energy"
+			eventMap.value = cmd.scaledMeterValue
+			eventMap.unit = "kWh"
+			events += createEvent(eventMap)
+		} else if (cmd.scale == 0x02) {
+			eventMap.name = "power"
+			eventMap.value = Math.round(cmd.scaledMeterValue)
+			eventMap.unit = "W"
+			events += createEvent(eventMap)
+			if (cmd.scaledMeterValue) {
+				events += createEvent([name: "windowShade", value: state.blindsLastCommand])
+				events += [createEvent([name: "level", value: state.shadeTarget, displayed: false]), createEvent([name: "shadeLevel", value: state.shadeTarget, displayed: false])]
```

Comment:
the device sends a power report (but not a level report) when it finishes moving?

Reply:
Yes, I have to trigger level reports manually. If device sends them by itself, most times it's so late, that user would get a timeout error before we can actually make an use of them.

Reply:
Apologies, I've mislead you a bit. Device doesn't send meter reports by itself either, they're triggered by the command here https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/36082/files#diff-549ffc27177cbae619e9ff6112545e3cR318 
When hub is added to 7th association group, it receives SwitchMultilevelSets, which are being sent every time motor starts and stops movement, when triggered by Z-Wave command, or every couple of seconds, when a position is being set by physical buttons.
I'm getting power reports first, instead of multilevel reports, to show user information whether blinds are going up or down.

Reply:
How does the power report help you show information about the blinds going up or down?

Reply:
Darn... To be more precise I should add "whether blinds are going up or down **or don't move at all**. 
If power report has value != 0, closing/opening event is sent (based on state.shadeTarget and current shadeLevel difference). Otherwise, open/closed/partially open event is being created.

Reply:
ahh okay